app/models/assignment_repo.rb

```diff
@@ -45,9 +46,21 @@ def github_team_id
 
   # Public
   #
+  def assign_repo_name_suffix
+    unless @repo_name_suffix_number.present?
+      @repo_name_suffix_number = 1
+      return
+    end
+    @repo_name_suffix_number += 1
+  end
+
+  # Public
+  #
   def repo_name
     github_user = GitHubUser.new(user.github_client, user.uid)
-    "#{assignment.slug}-#{github_user.login(headers: GitHub::APIHeaders.no_cache_no_store)}"
+    base_name = "#{assignment.slug}-#{github_user.login(headers: GitHub::APIHeaders.no_cache_no_store)}"
+    return base_name unless @repo_name_suffix_number.present?
+    "#{base_name}-#{@repo_name_suffix_number}"
```

Comment:
What will happen when the suffix takes the name over the character limit for the GitHub API?